typespec/Applications.Core/environments.tsp

```diff
@@ -65,11 +65,44 @@ model EnvironmentProperties {
   @doc("Specifies Recipes linked to the Environment.")
   recipes?: Record<Record<RecipeProperties>>;
 
+  @doc("Specifies configurations needed for the recipes registered to the environment.")
+  recipeConfig?: RecipeConfigProperties;
+
   @doc("The environment extension.")
   @extension("x-ms-identifiers", [])
   extensions?: Array<Extension>;
 }
 
+@doc("Specifies recipe configurations needed for the recipes.")
+model RecipeConfigProperties {
+  @doc("Specifies the terraform config properties")
```

Comment:
Configuration for Terraform Recipes. Controls how Terraform plans and applies templates as part of Recipe deployment.